### PR TITLE
Add custom cop

### PR DESCRIPTION
**Pull request by dfaulken**
Use `be` matcher rather than `eq` or `eql` for singleton types
(boolean, fixnum, etc).

Has support for autocorrect, which I used here.

Next up: preferring predicate matchers to matching against the return value of a predicate method.

### CODE REVIEW

File: lib/custom_cops/be_for_singleton.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module RuboCop
+  module Cop
+    module CustomCops
+      class BeForSingleton < Cop
```
Comment (dfaulken) [inline lib/custom_cops/be_for_singleton.rb]: Fully open to suggestions for better names.

Comment (werebus) [inline lib/custom_cops/be_for_singleton.rb]: Yeah, is "singleton" technically correct for `Fixnum`s and `Float`s?

Comment (dfaulken) [inline lib/custom_cops/be_for_singleton.rb]: No! I couldn't think of what it *is* called though.

Comment (werebus) [inline lib/custom_cops/be_for_singleton.rb]: "nonduplicable", perhaps?

https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/object/duplicable.rb

Comment (dfaulken) [inline lib/custom_cops/be_for_singleton.rb]: I like that better.

**Comment by dfaulken** [comment]
Pulling this down in favor of getting umts/custom-cops working.

**Pull request closed without merging**